static/app/views/preprod/header/buildDetailsHeaderContent.tsx

```diff
@@ -0,0 +1,133 @@
+import styled from '@emotion/styled';
+
+import {addErrorMessage} from 'sentry/actionCreators/indicator';
+import {Breadcrumbs, type Crumb} from 'sentry/components/breadcrumbs';
+import {Alert} from 'sentry/components/core/alert';
+import {Button} from 'sentry/components/core/button';
+import Placeholder from 'sentry/components/placeholder';
+import {IconEllipsis, IconTelescope} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {BuildDetailsApiResponse} from 'sentry/views/preprod/types';
+
+type BuildDetailsHeaderError = {error: string; status: 'error'};
+type BuildDetailsHeaderLoading = {status: 'loading'};
+type BuildDetailsHeaderSuccess = {
+  buildDetails: BuildDetailsApiResponse;
+  status: 'success';
+};
+
+export type BuildDetailsHeaderContentProps =
+  | BuildDetailsHeaderError
+  | BuildDetailsHeaderLoading
+  | BuildDetailsHeaderSuccess;
+
+export function BuildDetailsHeaderContent(props: BuildDetailsHeaderContentProps) {
+  const {status} = props;
+
+  if (status === 'loading') {
+    return (
+      <HeaderContainer>
+        <Placeholder height="20px" width="200px" style={{marginBottom: space(2)}} />
+        <HeaderContent>
+          <Title>
+            <Placeholder height="32px" width="300px" />
+          </Title>
+          <Actions>
+            <Placeholder height="32px" width="120px" style={{marginRight: space(1)}} />
+            <Placeholder height="32px" width="40px" />
+          </Actions>
+        </HeaderContent>
+      </HeaderContainer>
+    );
+  }
+
+  if (status === 'error') {
```

Comment:
Not sure what we'd even show in the case of an error so I think we might want to hoist these states up to the top-level buildDetails page, but can do in a later PR